examples/servers/simple-auth/mcp_simple_auth/auth_server.py

```diff
@@ -0,0 +1,259 @@
+
+from .github_oauth_provider import GitHubOAuthProvider, GitHubOAuthSettings
+
+logger = logging.getLogger(__name__)
+
+
+class AuthServerSettings(GitHubOAuthSettings):
+    """Settings for the Authorization Server."""
+
+    model_config = SettingsConfigDict(env_prefix="MCP_")
+
+    # Server settings
+    host: str = "localhost"
+    port: int = 9000
+    server_url: AnyHttpUrl = AnyHttpUrl("http://localhost:9000")
+    github_callback_path: str = "http://localhost:9000/github/callback"
+
+    def __init__(self, **data):
+        """Initialize settings with values from environment variables."""
+        super().__init__(**data)
+
+
+class GitHubProxyAuthProvider(GitHubOAuthProvider):
+    """
+    Authorization Server provider that proxies GitHub OAuth.
+
+    This provider:
+    1. Issues MCP tokens after GitHub authentication
+    2. Stores token state for introspection by Resource Servers
+    3. Maps MCP tokens to GitHub tokens for API access
+    """
+
+    def __init__(self, settings: AuthServerSettings):
+        super().__init__(settings, settings.github_callback_path)
+
+
+def create_authorization_server(settings: AuthServerSettings) -> Starlette:
+    """Create the Authorization Server application."""
+    oauth_provider = GitHubProxyAuthProvider(settings)
+
+    auth_settings = AuthSettings(
+        issuer_url=settings.server_url,
+        client_registration_options=ClientRegistrationOptions(
+            enabled=True,
+            valid_scopes=[settings.mcp_scope],
+            default_scopes=[settings.mcp_scope],
+        ),
+        required_scopes=[settings.mcp_scope],
+        authorization_servers=None,
+    )
+
+    # Create OAuth routes
+    routes = create_auth_routes(
+        provider=oauth_provider,
+        issuer_url=auth_settings.issuer_url,
+        service_documentation_url=auth_settings.service_documentation_url,
+        client_registration_options=auth_settings.client_registration_options,
+        revocation_options=auth_settings.revocation_options,
+    )
+
+    # Add GitHub callback route
+    async def github_callback_handler(request: Request) -> Response:
+        """Handle GitHub OAuth callback."""
+        code = request.query_params.get("code")
+        state = request.query_params.get("state")
+
+        if not code or not state:
+            raise HTTPException(400, "Missing code or state parameter")
+
+        try:
+            redirect_uri = await oauth_provider.handle_github_callback(code, state)
+            return RedirectResponse(url=redirect_uri, status_code=302)
+        except HTTPException:
+            raise
+        except Exception as e:
+            logger.error("Unexpected error", exc_info=e)
+            return JSONResponse(
+                status_code=500,
+                content={
+                    "error": "server_error",
+                    "error_description": "Unexpected error",
+                },
+            )
```

Comment:
This is not necessary. It will already respond with a 500, and you'll be able to see in the logs.